workshops/models.py

```diff
@@ -296,17 +296,20 @@ def get_by_ident(ident):
         YYYY-MM-DD-site slug, and indicates an event whose dates are firm.
         The real indicator is the 'published' flag.
         '''
-
-        # match event ID
-        if re.match(r'^\d+$', ident):
-            return Event.objects.get(id=ident)
-
-        # match event slug in format YYYY-MM-**** (ie. this works for
-        # YYYY-MM-DD-**** too)
-        if re.match(r'^\d{4}-\d{2}-.+$', ident):
-            return Event.objects.get(slug=ident)
-
-        raise ObjectDoesNotExist(ident)
+        kwargs = {}
+        try:
+            # first try to match using ID
+            kwargs["id"] = int(ident)
+        except ValueError:
+            # match event slug in format YYYY-MM-****
+            #(ie. this works for YYYY-MM-DD-**** too)
```

Comment:
Nit-picking: if we decide to go with this method, there's a space missing at the beginning of this comment. It should look like:
```python
            # (ie. this works for …
```